### PR TITLE
Extended public API

### PR DESCRIPTION
**Pull request by dkozinski**
Added function that sets the value for a given param from XEVE_PARAM struct

### CODE REVIEW

File: src_base/xeve_param_parse.c [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+#define OFFSET(x) offsetof(XEVE_PARAM, x)
+
+#define SET_XEVE_PARAM_METADATA( param, data_type) \
+    { .name=#param, .type=data_type, .offset=OFFSET(param) }
+
+/* Type of data stored by a given param from XEVE_PARAM stuct */
+typedef enum DataType {
+    DT_INTEGER    = (1 << 0), /* integer type value */
+    DT_DOUBLE     = (1 << 1), /* integer type value */
```
Comment (dariusz-f) [inline src_base/xeve_param_parse.c:51]: in comment should be "double type value"

File: src_base/xeve_param_parse.c [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    SET_XEVE_PARAM_METADATA( rpl1,                                      DT_STRING), // char rpl1[32][256];
+
+    SET_XEVE_PARAM_METADATA( rpls_l0_cfg_num,                           DT_INTEGER ),
+    SET_XEVE_PARAM_METADATA( rpls_l1_cfg_num,                           DT_INTEGER ),
+
+    /* preset parameter */
+    SET_XEVE_PARAM_METADATA( ats_intra_fast,                            DT_INTEGER ),
+    SET_XEVE_PARAM_METADATA( me_fast,                                   DT_INTEGER ),
+
+    // /* VUI options*/
```
Comment (dariusz-f) [inline src_base/xeve_param_parse.c:210]: fix comment style

File: src_base/xeve_param_parse.c [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    SET_XEVE_PARAM_METADATA( max_bits_per_mb_denom,                     DT_INTEGER ),
+    SET_XEVE_PARAM_METADATA( log2_max_mv_length_horizontal,             DT_INTEGER ),
+    SET_XEVE_PARAM_METADATA( log2_max_mv_length_vertical,               DT_INTEGER ),
+    SET_XEVE_PARAM_METADATA( num_reorder_pics,                          DT_INTEGER ),
+    SET_XEVE_PARAM_METADATA( max_dec_pic_buffering,                     DT_INTEGER ),
+    SET_XEVE_PARAM_METADATA( aspect_ratio_info_present_flag,            DT_INTEGER ),
+    SET_XEVE_PARAM_METADATA( video_signal_type_present_flag,            DT_INTEGER ),
+    SET_XEVE_PARAM_METADATA( colour_description_present_flag,           DT_INTEGER ),
+
+    // /* SEI options*/
```
Comment (dariusz-f) [inline src_base/xeve_param_parse.c:246]: fix comment style

File: src_base/xeve_param_parse.c [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+            // If PARAM_STR_MAX_LEN is less than or equal to the length of val,
+            // a null character (\0) is not appended to the copied string (char*)(args->opts[idx].opt_storage)
+            // The line below prevents truncation of destination string to not-null terminated string
+            ((char*)((void*)params + param_meta->offset))[PARAM_STR_MAX_LEN-1] = 0;
+
+            break;
+        default:
+            return XEVE_ERR;
+    }
```
Comment (dariusz-f) [inline src_base/xeve_param_parse.c:306]: where is DT_DOUBLE case?

**Review by dariusz-f — CHANGES_REQUESTED** [review]

File: src_base/xeve_param_parse.c [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+#define PARAMS_END_KEY                      (0)
+#define PARAM_STR_MAX_LEN                   (256)
+
+#define OFFSET(x) offsetof(XEVE_PARAM, x)
+
+#define SET_XEVE_PARAM_METADATA( param, data_type) \
+    { .name=#param, .type=data_type, .offset=OFFSET(param) }
+
+/* Type of data stored by a given param from XEVE_PARAM stuct */
+typedef enum DataType {
```
Comment (washish) [inline src_base/xeve_param_parse.c:49]: Can you replace these variable names to ALL CAPS - underscore format?

File: src_base/xeve_param_parse.c [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+
+            break;
+        case DT_STRING:
+
+            strncpy((char*)((void*)params + param_meta->offset), value, PARAM_STR_MAX_LEN);
+
+            // If PARAM_STR_MAX_LEN is less than or equal to the length of val,
+            // a null character (\0) is not appended to the copied string (char*)(args->opts[idx].opt_storage)
+            // The line below prevents truncation of destination string to not-null terminated string
+            ((char*)((void*)params + param_meta->offset))[PARAM_STR_MAX_LEN-1] = 0;
```
Comment (washish) [inline src_base/xeve_param_parse.c:301]: Should it be '\0' instead of 0?

Comment (dkozinski) [reply src_base/xeve_param_parse.c:301]: That's the same

File: src_base/xeve_param_parse.c (hunk 5 above)
Comment (dkozinski) [reply src_base/xeve_param_parse.c:49]: of course, I can

**Review by dariusz-f — APPROVED** [review]

**Review by washish — APPROVED** [review]

File: inc/xeve.h [hunk 7]
```diff
@@ -594,6 +594,13 @@ int  XEVE_EXPORT xeve_param_default(XEVE_PARAM* param);
 int  XEVE_EXPORT xeve_param_ppt(XEVE_PARAM* param, int profile, int preset, int tune);
 int  XEVE_EXPORT xeve_param_check(const XEVE_PARAM* param);
 
+/** Set one parameter by name.
+ * @retval 0 on success
+ * @retval XEVE_ERR_INVALID_ARGUMENT
+ * @retval XEVE_ERR
+ */
```
Comment (kpchoi) [inline inc/xeve.h:601]: Currently, we're not using doxygen syntax. How about removing this comments for consistency?

Comment (kpchoi) [reply inc/xeve.h:601]: My recommendation is only for API, not internal comments.

Comment (dkozinski) [reply inc/xeve.h]: @kpchoi 

If everything is OK please merge the PR to the main branch.

**Review by kpchoi — APPROVED** [review]
LGTM

**Pull request merged**